unit-tests/func/serializable-device/test-l500-json-load.py

```diff
@@ -60,13 +60,38 @@
 }
 """
 
+def log_settings_differences():
+    global data, depth_sensor, sd
+    depth_sensor.set_option(rs.option.visual_preset, int(rs.l500_visual_preset.low_ambient_light))
+    actual_data = str( sd.serialize_json() )
+    log.debug_indent()
+    try:
+        # logging the differences in the settings between the expected and the actual values
+        log.d( "Printing differences between expected and actual settings values:" )
+        for expected_line, actual_line in zip( data.splitlines()[2:-1], actual_data.splitlines()[1:-1] ):
```

Comment:
What happens when they're a different size because a new option was added?

Reply:
The comparisons will stop when the shorter list is done

Reply:
But they'll be in the same order??

Reply:
if the added option is at the beginning everything will be shifted and we'll get a failure and see the new option

Reply:
But then the output (expected vs found) would include all the lines...
Can we count on the options being in the same order, even if an option wasn't added?
I think we need to convert to dictionaries and compare apples to apples?